File: src/eqn/ast/EqnAstNodeCos.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

public class EqnAstNodeCos extends EqnAstNodePreDefinedUnaryFunction {

    public EqnAstNodeCos(@NotNull EqnAstNode equationNode) throws EqnException {
        super("cos", equationNode);
    }

    @Override
    public double evaluate() throws EqnException {
        return Math.cos(operand().evaluate());
    }
}

File: src/eqn/ast/EqnAstNodeCos.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary
import eqn.parser.exception.EqnException
import kotlin.math.cos

class EqnAstNodeCos(operand: EqnAstNode) : EqnAstNodeUnary("cos", Type.PreDefinedUnaryFunction, PrecedenceType.Function, operand) {

    @Throws(EqnException::class)
    override fun evaluate(arguments: Map<String, Double>?): Double {
        return cos(operand.evaluate(arguments))
    }
}